### DISCUSSION TITLE
Is there any way to name test runs with httpexpect?

### DISCUSSION THREAD

**Discussion by cyantarek** [gavv/httpexpect#151] [Q&A] [answered]
Hi,

Currently I'm doing end to end test with httpexpect. There are currently 100+ sequential url calls in the test. The problem is, when test failed, it just shows line number in the source code. If I can give name to test calls (in this case, single url request call), and httpexpect would print the name when failed or passed along side with source code line number, it would be better.

**Comment by gavv** [comment]
Hi,

There is no way to do it currently.

It would be a nice addition probably. We could add `name` field to `chain` struct and a  `WithName()` setter for it to `Request` struct (request contains a chain). The chain object is propagated to all nested objects and is used to report all errors, so it should be enought to report name in `chain.fail()` and that's all I hope. Pull requests are welcome :)

But maybe you can use Go subtests instead? Are there any problems with this approach?

**Comment by gavv** [comment, marked as answer]
It's now possible. You can pass `Config.TestName` as described here: https://github.com/gavv/httpexpect#custom-config

Also, when you're using `httpexpect.Default()`, `TestName` will be set automatically.

Also you can set `Request.WithName()` to specify request name.
